data-protocols/dsp/dsp-transfer-process/dsp-transfer-process-api/src/main/java/org/eclipse/edc/protocol/dsp/transferprocess/api/controller/DspTransferProcessApiController.java

```diff
@@ -0,0 +1,223 @@
+        this.monitor = monitor;
+        this.protocolService = protocolService;
+        this.registry = registry;
+        this.mapper = typeManager.getMapper(TYPE_MANAGER_CONTEXT_JSON_LD);
+        this.identityService = identityService;
+        this.dspCallbackAddress = dspCallbackAddress;
+    }
+    
+    /**
+     * Retrieves an existing transfer process. This functionality is not yet supported.
+     *
+     * @param id the ID of the process
+     * @return the transfer process in JSON-LD expanded form
+     */
+    @GET
+    @Path("/{id}")
+    public JsonObject getTransferProcess(@PathParam("id") String id) {
+        monitor.debug(format("DSP: Incoming request for transfer process with id %s", id));
+        
+        throw new UnsupportedOperationException("Getting a transfer process not yet supported.");
+    }
+    
+    /**
+     * Initiates a new transfer process that has been requested by the counter-party.
+     *
+     * @param jsonObject the {@link TransferRequestMessage} in JSON-LD expanded form
+     * @param token the authorization header
+     * @return the created transfer process  in JSON-LD expanded form
+     */
+    @POST
+    @Path(TRANSFER_INITIAL_REQUEST)
+    public Map<String, Object> initiateTransferProcess(JsonObject jsonObject, @HeaderParam(HttpHeaders.AUTHORIZATION) String token) {
+        monitor.debug("DSP: Incoming TransferRequestMessage for initiating a transfer process");
+        
+        var claimToken = checkAuthToken(token);
+        
+        var message = registry.transform(expand(jsonObject).getJsonObject(0), TransferRequestMessage.class)
+                .orElseThrow(failure -> new InvalidRequestException(format("Failed to read request body: %s", failure.getFailureDetail())));
+        
+        var transferprocess = protocolService.notifyRequested(message, claimToken)
+                .orElseThrow(failure -> new EdcException(format("TransferProcess could not be initiated: %s", failure.getFailureDetail())));
+        
+        var transferProcessJson = registry.transform(transferprocess, JsonObject.class)
+                .orElseThrow(failure -> new EdcException(format("Response could not be created: %s", failure.getFailureDetail())));
+        
+        return mapper.convertValue(compact(transferProcessJson, jsonLdContext()), Map.class);
+    }
+    
+    /**
+     * Notifies the connector that a transfer process has been started by the counter-part.
+     *
+     * @param id the ID of the process
+     * @param jsonObject the {@link TransferStartMessage} in JSON-LD expanded form
+     * @param token the authorization header
+     */
+    @POST
+    @Path("{id}" + TRANSFER_START)
+    public void transferProcessStart(@PathParam("id") String id, JsonObject jsonObject, @HeaderParam(HttpHeaders.AUTHORIZATION) String token) {
+        monitor.debug(format("DSP: Incoming TransferStartMessage for transfer process %s", id));
```

Comment:
ended up with a few more parameters, as I also added missing checks. But good call, this will definitely make it more maintainable in case changes are due